### PR TITLE
synthetic-chain package and CLI

### PR DESCRIPTION
**Pull request by turadg**
## Description

This simplifies proposal authoring and maintenance. It's a necessary refactor to support https://github.com/Agoric/agoric-sdk/issues/8605

Once this PR is landed, I'll follow up to,
1. make `@agoric/synthetic-chain` able to pull from A3P and add proposals on top of it
2. publish the package to NPM
3. use that NPM package to rework https://github.com/Agoric/agoric-sdk/pull/8635 
4. close https://github.com/Agoric/agoric-sdk/issues/8605

### CODE REVIEW

File: packages/synthetic-chain/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        "module": "NodeNext",
+        "moduleResolution": "nodenext",
+        "noEmit": true,
+        "target": "ESNext"
+    },
+    "include": [
+        "*.js",
+        "*.ts",
+        "proposals/**/*.js",
+        "proposals/**/*.ts",
```
Comment (dckc) [inline packages/synthetic-chain/tsconfig.json:16]: These don't seem to be included in `files` in `package.json`. Is that by design?

File: proposals/34:upgrade-10/yarn.lock [hunk 2]
```diff
@@ -5,6 +5,171 @@ __metadata:
...
+  resolution: "@agoric/synthetic-chain@file:./agoric-synthetic-chain-0.0.1-alpha.tgz#./agoric-synthetic-chain-0.0.1-alpha.tgz::hash=1144d8&locator=root-workspace-0b6124%40workspace%3A."
+  dependencies:
+    tsx: "npm:^3.12.8"
+  bin:
+    synthetic-chain: ./cli.ts
+  checksum: 012b67bb80fc818bdb63a16f05007c7183a63d45d0a99a2ce1477079fe2249eae0961c6325b30cea111f4dd38487011f20a94a1518a8b8badef7c4583992d131
+  languageName: node
+  linkType: hard
+
+"@esbuild/android-arm64@npm:0.18.20":
```
Comment (dckc) [inline proposals/34:upgrade-10/yarn.lock:19]: I see `esbuild` pop up in our builds from time to time. I wonder what depends on it.

Here's hoping for a moment to check.

File: proposals/34:upgrade-10/package.json [hunk 3]
```diff
@@ -8,6 +8,7 @@
     "type": "module",
     "license": "Apache-2.0",
     "dependencies": {
+        "@agoric/synthetic-chain": "file:./agoric-synthetic-chain-0.0.1-alpha.tgz",
```
Comment (dckc) [inline proposals/34:upgrade-10/package.json:11]: There's a copy of this tar file in each proposals directory? I guess that makes sense...

**Review by dckc — APPROVED** [review]
works for me

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -59,15 +59,18 @@ jobs:
...
       - run: npm install --global tsx
+      # Enable corepack for packageManager config
+      - run: corepack enable
+      - run: yarn install
 
       - name: build test images
         run: |
           docker info
-          ./buildTestImages.ts
+          node_modules/.bin/synthetic-chain build
```
[the remark this answers is not in the document]
Comment (turadg) [reply .github/workflows/ci.yml:71]: Yes but that's not working. I'll take another crack at it after the other items

File: packages/synthetic-chain/tsconfig.json (hunk 1 above)
Comment (turadg) [reply packages/synthetic-chain/tsconfig.json:16]: These globs are meant to get all files, independent of what's being published.

File: proposals/34:upgrade-10/yarn.lock (hunk 2 above)
Comment (turadg) [reply proposals/34:upgrade-10/yarn.lock:19]: 'tsx' does

File: proposals/34:upgrade-10/package.json (hunk 3 above)
Comment (turadg) [reply proposals/34:upgrade-10/package.json:11]: Once we're publishing to NPM then it can just pull from there. This is a stop gap

**Pull request merged**